jib-core/src/test/java/com/google/cloud/tools/jib/configuration/BuildConfigurationTest.java

```diff
@@ -215,90 +212,4 @@ public void testBuilder_missingValues() {
-              ContainerConfiguration.builder()
-                  .setProgramArguments(Arrays.asList("first", null))
-                  .build());
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Entrypoint element should not be null.
-    try {
-      BuildConfiguration.builder(Mockito.mock(JibLogger.class))
-          .setContainerConfiguration(
-              ContainerConfiguration.builder().setEntrypoint(Arrays.asList("first", null)).build());
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Exposed ports element should not be null.
-    try {
-      BuildConfiguration.builder(Mockito.mock(JibLogger.class))
-          .setContainerConfiguration(
-              ContainerConfiguration.builder()
-                  .setExposedPorts(Arrays.asList(new Port(1000, Protocol.TCP), null))
-                  .build());
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Labels element should not be null.
-    Map<String, String> badLabels = new HashMap<>();
-    badLabels.put("label-key", null);
-    try {
-      ContainerConfiguration.builder().setLabels(badLabels);
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Environment keys element should not be null.
-    Map<String, String> nullKeyMap = new HashMap<>();
-    nullKeyMap.put(null, "value");
-
-    try {
-      BuildConfiguration.builder(Mockito.mock(JibLogger.class))
-          .setContainerConfiguration(
-              ContainerConfiguration.builder().setEnvironment(nullKeyMap).build());
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Environment values element should not be null.
-    Map<String, String> nullValueMap = new HashMap<>();
-    nullValueMap.put("key", null);
-    try {
-      BuildConfiguration.builder(Mockito.mock(JibLogger.class))
-          .setContainerConfiguration(
-              ContainerConfiguration.builder().setEnvironment(nullValueMap).build());
-      Assert.fail("The IllegalArgumentException should be thrown.");
-    } catch (IllegalArgumentException ex) {
-      Assert.assertNull(ex.getMessage());
-    }
-
-    // Can accept empty environment.
-    BuildConfiguration.builder(Mockito.mock(JibLogger.class))
-        .setContainerConfiguration(
-            ContainerConfiguration.builder().setEnvironment(ImmutableMap.of()).build());
-
-    // Environment map can accept TreeMap and Hashtable.
-    BuildConfiguration.builder(Mockito.mock(JibLogger.class))
```

Comment:
These were removed, doesn't make sense to ensure `TreeMap`/`HashMap` are valid `Map` types.

Reply:
Now I remember: https://github.com/GoogleContainerTools/jib/issues/632#issue-341871451

The point of the test was to ensure that we don't call things like `containsKey(null)`, because while `ImmutableMap` is OK with that, some `Map` implementations like `TreeMap` throws an exception right away. Maybe adding the issue link as a comment make this clearer.

Reply:
that's a good point, I'll put them back in.